File: src/main/java/ninja/hudy/infosmog/repository/MeasurementRepository.java
package ninja.hudy.infosmog.repository;

import ninja.hudy.infosmog.model.Measurement;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;
import java.util.Set;

@Repository
public interface MeasurementRepository extends JpaRepository<Measurement, Long> {

    Optional<Measurement> findByStationIdAndTimestampAndTypeId(Long id, Long timestamp, Long typeId);

    Set<Measurement> findByStationIdAndTimestampAfter(Long id, Long timestamp);

    Set<Measurement> findByTimestampAfter(Long timestamp);
}

File: src/main/kotlin/pl/org/smok/infosmog/repository/MeasurementRepository.kt
package pl.org.smok.infosmog.repository

import pl.org.smok.infosmog.model.Measurement
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

import java.util.Optional

@Repository
interface MeasurementRepository : JpaRepository<Measurement, Long> {
    fun findByStationIdAndTimestampAndTypeId(id: Long?, timestamp: Long?, typeId: Long?): Optional<Measurement>

    fun findByStationIdAndTimestampAfter(id: Long?, timestamp: Long?): Set<Measurement>

    fun findByTimestampAfter(timestamp: Long?): Set<Measurement>
}
